contracts/alliance-nft-collection/src/tests/helpers.rs

```diff
@@ -61,24 +79,95 @@ pub fn query_nft(deps: Deps, token_id: &str) -> NftInfoResponse<Extension> {
+    from_json(&query(deps, mock_env(), msg).unwrap()).unwrap()
+}
 
-    TEMP_BALANCE
-        .save(deps.storage, &(contract_balance.amount - rewards))
-        .unwrap();
+pub fn claim_alliance_emissions(deps: &mut MockContext, rewards: Uint128) {
+    deps.querier
+        .bank_querier
+        .update_balance(MOCK_CONTRACT_ADDR, coins(rewards.u128(), ALLOWED_DENOM));
 
+    // CLAIM REWARDS
     let info = mock_info(MOCK_CONTRACT_ADDR, &[]);
     let env = mock_env();
-    let msg = ExecuteCollectionMsg::UpdateRewardsCallback {};
-    execute(deps, env, info, msg).unwrap();
+    let msg = ExecuteCollectionMsg::AllianceClaimRewards {};
+    let result = execute(deps.as_mut(), env.clone(), info.clone(), msg).unwrap();
+
+    assert_eq!(result.messages.len(), 1);
+    assert_eq!(
+        result.messages[0].msg,
+        CosmosMsg::Wasm(cosmwasm_std::WasmMsg::Execute {
+            contract_addr: MOCK_CONTRACT_ADDR.to_string(),
+            msg: to_json_binary(&ExecuteCollectionMsg::StakeRewardsCallback {}).unwrap(),
+            funds: vec![]
+        })
+    );
+
+    // STAKE REWARDS (in LST) CALLBACK
+    let msg = ExecuteCollectionMsg::StakeRewardsCallback {};
+    let result = execute(deps.as_mut(), env.clone(), info.clone(), msg).unwrap();
+
+    let previouse_balance = deps.querier.get_cw20_balance(MOCK_LST, MOCK_CONTRACT_ADDR);
+
+    let result_msg = ExecuteCollectionMsg::UpdateRewardsCallback(UpdateRewardsCallbackMsg {
+        previous_lst_balance: Uint128::new(previouse_balance),
+    });
+    assert_eq!(result.messages.len(), 2);
+    assert_eq!(
+        result.messages[0].msg,
+        Hub(Addr::unchecked("hub"))
+            .bond_msg(ALLOWED_DENOM, rewards.u128(), None)
+            .unwrap()
+    );
+    assert_eq!(
+        result.messages[1].msg,
+        CosmosMsg::Wasm(cosmwasm_std::WasmMsg::Execute {
+            contract_addr: MOCK_CONTRACT_ADDR.to_string(),
+            msg: to_json_binary(&result_msg).unwrap(),
+            funds: vec![]
+        })
+    );
+
+    // UPDATE REWARDS CALLBACK
+    // exchange rate of 1.2
+    let added_lst_amount = rewards.multiply_ratio(100u128, 120u128);
+    deps.querier.set_cw20_balance(
+        MOCK_LST,
+        MOCK_CONTRACT_ADDR,
+        previouse_balance + added_lst_amount.u128(),
+    );
+
+    let msg = result_msg;
+    let result = execute(deps.as_mut(), env, info, msg).unwrap();
+
+    let treasury_amount = Decimal::percent(10) * added_lst_amount;
+    let rewards = added_lst_amount - treasury_amount;
+
+    assert_eq!(result.messages.len(), 1);
+    assert_eq!(
+        result.messages[0].msg,
+        AssetInfo::cw20(Addr::unchecked(MOCK_LST))
+            .with_balance(treasury_amount)
+            .transfer_msg(MOCK_DAO_TREASURY_ADDRESS)
+            .unwrap()
+    );
+    assert_eq!(
+        result.attributes,
+        vec![
+            attr("action", "update_rewards_callback"),
+            attr("rewars_collected", rewards),
```

Comment:
Seems like this test is failing due the following typo:
```suggestion
            attr("rewards_collected", rewards),
```